compiler/statements.go

```diff
@@ -282,7 +282,7 @@ func (c *funcContext) translateStmt(stmt ast.Stmt, label *types.Label) {
 			}
 			results = c.resultNames
 		}
-		c.Printf("return%s;", c.translateResults(results))
+		c.Printf("return%s;", c.translateResults(results, c.Flattened[s]))
```

Comment:
Seems like the new logic is only used here. Can we move it above this line, e.g. `rVal := c.translateResults(results); if c.Flattened[s] { [...] }`? I'd rather avoid having `c.Printf` in the helper function `translateResults`.